pennylane/ops/op_math/prod.py

```diff
@@ -276,20 +286,31 @@ def diagonalizing_gates(self):
         Returns:
             list[.Operator] or None: a list of operators
         """
-
-        eigen_vectors = self.eigendecomposition["eigvec"]
-        return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        if self.overlapping_wires:
+            eigen_vectors = self.eigendecomposition["eigvec"]
+            return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        diag_gates = []
+        for factor in self.factors:
+            diag_gates.extend(factor.diagonalizing_gates())
+        return [qml.adjoint(gate) for gate in diag_gates]
```

Comment:
Why adjoint?

Reply:
I was confused, because in the case where there are overlapping wires, you take the adjoint of the eigenvectors. But in this case we should NOT do it. 

Removing the adjoints...